👨‍💼 Sprint 4 – Time Series Analysis & Business Trends
🏢 Business Scenario

The CEO walks into the meeting and says:

"I know our total revenue is $2.3M.
But I don't know how the business has grown over time.
Show me the trend."

As a Data Analyst, your job is no longer to calculate totals—it's to identify patterns and trends.

📌 Part 1 – Explore the Dates
Business Requirement

"Before analyzing trends, understand the available time period."

Task 1

Find the first and last order date.

Think:

How many years of data do we have?

Is it enough for trend analysis?

In [0]:
%sql
SELECT
    MIN(`Order Date`) AS Earliest_Order_Date,
    MAX(`Order Date`) AS Latest_Order_Date
FROM abc_retail_pvt_ltd.data_analytics.superstore;

Earliest_Order_Date,Latest_Order_Date
2023-01-03T00:00:00.000Z,2026-12-30T00:00:00.000Z


The year from 2023 to 2026 so 4 years of data were available to be analyzed and know the business status

The dataset contains four years of historical sales data (2023–2026), providing sufficient information to analyze sales trends, customer behavior, and business performance over time.

Task 2

How many orders were placed each year?

Business Question:

Is the company growing every year?

In [0]:
%sql
select year(`Order Date`), count(distinct `Order ID`) from abc_retail_pvt_ltd.data_analytics.superstore
group by year(`Order Date`)
order by year(`Order Date`);

year(Order Date),count(DISTINCTOrder ID)
2023,995
2024,1053
2025,1340
2026,1723


The number of customer orders increased consistently each year, growing from 2051 orders in 2023 to 3379 orders in 2026. This indicates positive business growth and increased customer demand.

In [0]:
%sql
select year(`Order Date`), Round ((sum(`Profit`)/sum(`Sales`))*100,2) as Profit_margin 
from abc_retail_pvt_ltd.data_analytics.superstore
group by year(`Order Date`)
order by year(`Order Date`);

year(Order Date),Profit_margin
2023,10.46
2024,13.11
2025,13.46
2026,12.87


Profit margin slightly declined from 13.46% to 12.87% in 2026, suggesting that although sales increased, profitability did not improve at the same rate.

Task 3

Revenue by Year

Business Question:

Which year generated the highest revenue?

In [0]:
%sql
select year(`Order Date`), round(sum(`Sales`),2) as Total_Revenue 
from abc_retail_pvt_ltd.data_analytics.superstore
group by year(`Order Date`)
order by total_revenue desc;


year(Order Date),Total_Revenue
2026,745567.53
2025,613933.58
2023,494040.21
2024,472993.03


Revenue increased year over year, with 2026 recording the highest annual revenue, indicating sustained business growth.

Task 4

Profit by Year

Business Question:

Which year generated the highest profit?

In [0]:
%sql
select year(`Order Date`), round(sum(`Profit`),2) as Total_Profit 
from abc_retail_pvt_ltd.data_analytics.superstore
group by year(`Order Date`)
order by total_profit desc;

year(Order Date),Total_Profit
2026,95926.35
2025,82665.20
2024,62020.97
2023,51684.30


2026 generated the highest annual profit of $95,926, indicating improved financial performance compared to previous years.

📌 Part 2 – Monthly Analysis

Now the Sales Director asks:

"Show me monthly trends."

Task 5

Monthly Revenue

⚠️ Remember our previous discussion.

Don't group only by month.

Think carefully.

In [0]:
%sql
select year(`Order Date`), Month(`Order Date`), Round(sum(`Sales`),2) as Monthy_Profile
from abc_retail_pvt_ltd.data_analytics.superstore
group by year(`Order Date`), Month(`Order Date`)
order by year(`Order Date`), Month(`Order Date`);

year(Order Date),Month(Order Date),Monthy_Profile
2023,1,14518.06
2023,2,4519.89
2023,3,56933.91
2023,4,28295.34
2023,5,26319.77
2023,6,34669.48
2023,7,33946.39
2023,8,28918.34
2023,9,82670.43
2023,10,32413.34


Finding

Sales consistently increase during November and December.

Possible Reason

Holiday shopping and year-end promotions.

Recommendation

Increase inventory and marketing before Q4.

Task 7

Monthly Orders

In [0]:
%sql
select year(`Order Date`), Month(`Order Date`), count(`Order ID`) as Monthy_orders
from abc_retail_pvt_ltd.data_analytics.superstore
group by year(`Order Date`), Month(`Order Date`)
order by year(`Order Date`), Month(`Order Date`);

year(Order Date),Month(Order Date),Monthy_orders
2023,1,84
2023,2,46
2023,3,159
2023,4,135
2023,5,132
2023,6,136
2023,7,143
2023,8,158
2023,9,273
2023,10,168


Task 8

Monthly Average Order Value

In [0]:
%sql
select year(`Order Date`), Month(`Order Date`), round(sum(`Sales`) / count(distinct `Order ID`),2) as AOV 
from abc_retail_pvt_ltd.data_analytics.superstore
group by year(`Order Date`), Month(`Order Date`)
order by year(`Order Date`), Month(`Order Date`);

year(Order Date),Month(Order Date),AOV
2023,1,439.94
2023,2,161.42
2023,3,790.75
2023,4,428.72
2023,5,365.55
2023,6,517.45
2023,7,522.25
2023,8,385.58
2023,9,616.94
2023,10,400.16


📌 Part 3 – Quarterly Analysis

Business Question:

Which quarter performs best?

Find:

Revenue
Profit
Orders

for every quarter.

In [0]:
%sql
SELECT
    YEAR(`Order Date`) AS Year,
    CONCAT('Q', QUARTER(`Order Date`)) AS Quarter,
    ROUND(SUM(`Sales`), 2) AS Revenue,
    ROUND(SUM(Profit), 2) AS Profit,
    COUNT(DISTINCT `Order ID`) AS Orders
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY YEAR(`Order Date`), QUARTER(`Order Date`)
ORDER BY YEAR(`Order Date`), QUARTER(`Order Date`);

Year,Quarter,Revenue,Profit,Orders
2023,Q1,75971.86,4095.15,133
2023,Q2,89284.59,11684.99,205
2023,Q3,145535.16,13517.37,274
2023,Q4,183248.60,22386.79,383
2024,Q1,70391.95,9554.66,148
2024,Q2,89239.45,12200.19,215
2024,Q3,130449.38,16880.30,277
2024,Q4,182912.25,23385.82,413
2025,Q1,94840.22,11628.49,183
2025,Q2,136809.14,16594.68,299


Q4 generated the highest revenue in every year.

Possible Reason

Holiday sales, Black Friday, Christmas promotions, and year-end purchasing behavior likely contributed to increased sales.

Recommendation

Allocate additional inventory, logistics, and marketing budget during Q4 to maximize revenue.

Revenue increased consistently across all four years, reaching its highest value in 2026.

This indicates sustained business growth rather than a one-time increase.

Management should continue investing in high-performing categories while analyzing the factors contributing to annual growth.

# Business Requirement
Identify the year with the highest revenue.

# Approach
Group sales by year using the Order Date column and calculate total revenue.

# SQL
SELECT ...

# Output
2026 → $733,215

# Finding
2026 recorded the highest annual revenue.

# Business Insight
Revenue increased consistently across the four-year period, indicating sustained business growth.

# Recommendation
Investigate the key drivers behind 2026's performance and replicate successful strategies in future years.